### ISSUE TITLE
addOption type

### ISSUE THREAD

**Issue by nu8** [dart-lang/args#154]
If I have a file like this:

~~~dart
import 'package:args/args.dart';

void main() {
   var o = ArgParser();
   o.addOption('length', defaultsTo: 1);
}
~~~

I get this result:

~~~
Error: The argument type 'int' can't be assigned to the parameter type 'String'.
~~~

are we only allowed to use String arguments? With other languages, you can set
a type. For example with Go:

~~~go
package main
import "flag"

func main() {
   var n int
   flag.IntVar(&n, "length", 1, "substring length")
   flag.Parse()
}
~~~

**Comment by nu8** [comment]
workaround:

~~~dart
import 'package:args/args.dart';

void main(a) {
   var o = ArgParser();
   o.addOption('length', defaultsTo: '1');
   var m = o.parse(a);
   var n_len = int.parse(m['length']);
}
~~~

**Comment by natebosch** [comment]
> are we only allowed to use String arguments?

Yes. What we start with at the process level is `List<String>` and so we only handle `String` options and `bool` flags (exists or not). We don't attempt type coercion from `String` to other types like `int`. If there is enough demand we could add an `addIntOption` method, but it would be a breaking change so we'd need a strong argument in favor. So far the few use cases we've seen use the `int.parse` approach you've taken here.

**Comment by natebosch** [comment]
We could also maybe do something like `addParsedOption<T>(String name, T Function(String) parse, {T? defaultsTo})`...
